cmd/prometheus/main.go

```diff
@@ -670,37 +672,37 @@ func main() {
 	// they are not specific to an SD instance.
 	err = discovery.RegisterK8sClientMetricsWithPrometheus(prometheus.DefaultRegisterer)
 	if err != nil {
-		level.Error(logger).Log("msg", "failed to register Kubernetes client metrics", "err", err)
+		logger.Error("failed to register Kubernetes client metrics", "err", err)
 		os.Exit(1)
 	}
 
 	sdMetrics, err := discovery.CreateAndRegisterSDMetrics(prometheus.DefaultRegisterer)
 	if err != nil {
-		level.Error(logger).Log("msg", "failed to register service discovery metrics", "err", err)
+		logger.Error("failed to register service discovery metrics", "err", err)
 		os.Exit(1)
 	}
 
-	discoveryManagerScrape = discovery.NewManager(ctxScrape, log.With(logger, "component", "discovery manager scrape"), prometheus.DefaultRegisterer, sdMetrics, discovery.Name("scrape"))
+	discoveryManagerScrape = discovery.NewManager(ctxScrape, logger.With("component", "discovery manager scrape"), prometheus.DefaultRegisterer, sdMetrics, discovery.Name("scrape"))
 	if discoveryManagerScrape == nil {
-		level.Error(logger).Log("msg", "failed to create a discovery manager scrape")
+		logger.Error("failed to create a discovery manager scrape")
 		os.Exit(1)
 	}
 
-	discoveryManagerNotify = discovery.NewManager(ctxNotify, log.With(logger, "component", "discovery manager notify"), prometheus.DefaultRegisterer, sdMetrics, discovery.Name("notify"))
+	discoveryManagerNotify = discovery.NewManager(ctxNotify, logger.With("component", "discovery manager notify"), prometheus.DefaultRegisterer, sdMetrics, discovery.Name("notify"))
 	if discoveryManagerNotify == nil {
-		level.Error(logger).Log("msg", "failed to create a discovery manager notify")
+		logger.Error("failed to create a discovery manager notify")
 		os.Exit(1)
 	}
 
 	scrapeManager, err := scrape.NewManager(
 		&cfg.scrape,
-		log.With(logger, "component", "scrape manager"),
-		func(s string) (log.Logger, error) { return logging.NewJSONFileLogger(s) },
+		logger.With("component", "scrape manager"),
+		func(s string) (*logging.JSONFileLogger, error) { return logging.NewJSONFileLogger(s) },
```

Comment:
Should this return a more abstract interface type?

Reply:
I can be convinced of the usefulness there. 

The previous "raison d'être" for the json file logger, as far as I can tell, was to satisfy the `promql.QueryLogger` interface:
https://github.com/prometheus/prometheus/pull/14906/files#diff-ad2761bb162a2f12326b18f2714ef54f4cd476cfe3388f77b2b20c5ff82e4faaL127-L130

Does that feel more appropriate for our usage here?